### PR TITLE
Add world npc

### PR DESCRIPTION
**Pull request by MaxMorrison6**
Created the World and Npc files and added basic attributes to Location, Tile, and Region files

### CODE REVIEW

File: Location.java [hunk 1]
```diff
@@ -4,7 +4,14 @@
 
 public final class Location {
 
-    private static Tile tile;
-    private static Region region;
+    private Tile tile;
+    private Region region;
+
+    public Location(Tile tile, Region region){
```
Comment (JacksonKolb) [inline Location.java]: add a space after region, and remove the empty lines after this.region

File: Region.java [hunk 2]
```diff
@@ -7,5 +7,16 @@
...
+    private ArrayList<Tile> tileList;
+    private ArrayList<Region> neighborList;
+
+    public Region(String regionName, ArrayList tileList, ArrayList neighborList){
+
+        this.regionName = regionName;
+        this.tileList = tileList;
+        this.neighborList = neighborList;
+
+    }
```
Comment (JacksonKolb) [inline Region.java]: delete this nephew (The empty line)

File: Tile.java [hunk 3]
```diff
@@ -1,8 +1,24 @@
...
 
 
 public final class Tile {
 
+    private Region tileRegion;
+    private boolean traversable;
+    private boolean border;
+    private ArrayList<Tile> neighborList;
+
+    public Tile(Region tileRegion, boolean traversable, boolean border ArrayList neighborList){
```
Comment (JacksonKolb) [inline Tile.java]: same here

**Comment by MaxMorrison6** [comment]
Updated PR with fixed formatting

**Comment by JacksonKolb** [comment]
10/10 would review again

**Pull request merged**